### PR TITLE
Few things

### PR DESCRIPTION
**Pull request by mirek**
Two things - title and `clear!`.

Clear can be called after progress bar is finished to clear progress bar line completely.

Title is optional, if set, it'll be shown on the left.

Added example with steps.

### CODE REVIEW

File: lib/progress_bar.rb [hunk 1]
```diff
@@ -73,19 +85,24 @@ def eta
 
   def to_s
     self.count = max if count > max
-    meters.inject("") do |text, meter|
-      text << render(meter) + " "
-    end.strip
+    meters
+      .map { |meter| render(meter) }
+      .reject { |rendered_meter| rendered_meter.nil? }
```
Comment (paul) [inline lib/progress_bar.rb]: This could be replaced with `#compact`.

File: examples/steps.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+
+require File.expand_path(File.join(File.dirname(__FILE__), '..', 'lib/progress_bar'))
+
+steps = [ 'foo.tar.gz2', 'bar.zip', 'baz.mp3' ]
+
+bar = ProgressBar.new
```
Comment (paul) [inline examples/steps.rb]: In this example, just re-initialize the bar each time in the loop:

```ruby
steps.each.with_index do |name, i|
  bar = Bar.new(name, 100)
  # ...
  bar.clear!
end
```

File: lib/progress_bar.rb [hunk 3]
```diff
@@ -73,19 +85,24 @@ def eta
...
+      .join ' '
+  end
+
+  def clear!
```
Comment (paul) [inline lib/progress_bar.rb]: How about we call it `#hide` or `#remove` instead?

File: lib/progress_bar.rb [hunk 4]
```diff
@@ -96,6 +113,10 @@ def width_of(meter)
     send(:"#{meter}_width")
   end
 
+  def render_title
+    "[#{title}]" if title
```
Comment (paul) [inline lib/progress_bar.rb]: I'm not sure I like how this looks when rendered. What about having it bare? Or a colon suffix?

```
My Title [########     ][...]
Other Title: [######        ][...]
```

I also wish there was a way to automagically align everything when this happens, but I don't see how to accomplish that without a specific `:title_width` option, which I'm not fond of...

File: spec/progress_bar_spec.rb [hunk 5]
```diff
@@ -18,6 +18,30 @@
     it { should == "[              ] [  0/100] [  0%] [00:10] [00:00] [  0.00/s]" }
   end
 
+  describe 'at count=0 with title' do
```
Comment (paul) [inline spec/progress_bar_spec.rb]: Wrap these tests in a `context "with title" do` block.

**Comment by paul** [comment]
Looks good, left some comments inline.

**Comment by mirek** [comment]
Thanks for quick response... will fix tomorrow.

**Comment by colstrom** [comment]
@mirek Any progress on those fixes, or anything I can help with if you're blocked? I'd love to make use of this feature.

**Comment by paul** [comment]
@mirek @colstrom If either of you want to fix this and update this PR, I'd be happy to merge it.

**Comment by paul** [comment]
Closing after 5 years with no updates.

**Pull request closed without merging**